tests/trainer/test_trainer.py

```diff
@@ -1444,3 +1445,64 @@ def test_trainer_profiler_incorrect_arg_type(profiler):
                        match=r"Only None, bool, str and subclasses of `BaseProfiler`"
                              r" are valid values for `Trainer`'s `profiler` parameter. *"):
         Trainer(profiler=profiler)
+
+
+@pytest.mark.parametrize(
+    ["limit_train_batches", "global_step", "num_training_batches", "current_epoch", "should_train"],
+    [pytest.param(0.2, 0, 0, 0, False), pytest.param(0.5, 10, 2, 4, True)],
+)
+def test_disabled_training_for_insufficient_limit_train_batches(tmpdir, limit_train_batches, global_step,
```

Comment:
maybe update `test_disabled_training` in the same file accordingly?

Reply:
With the similar params? Sure!

Reply:
yes. you can include one more there with limit_train_batches=0 along with others.

Reply:
I think it would be better to have it separate as `limit_train_batches=0` would be independent of `dataset_len` and `batch_size`.

Reply:
ok yeah, no need to include one with `limit_train_batches=0` since it's changed now. you can either remove the old test or update that with this one.

Reply:
We shouldn't remove it as it tests for `fast_dev_run=True/False` switches.